dhall/src/Dhall/Deriving.hs

```diff
@@ -0,0 +1,212 @@
+{-# LANGUAGE AllowAmbiguousTypes  #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE PolyKinds            #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Dhall.Deriving
```

Comment:
I think the only thing really missing is a doctest and/or example illustrating how this works.  CI will require a module documentation header here anyway, so that's probably the best place to add it

Reply:
done!